### DISCUSSION TITLE
withoutOverlapping() not working properly

### DISCUSSION THREAD

**Discussion by AlbertoSinigaglia** [laravel/framework#33051] [General] [answered]
- Laravel Version: 6.x
- PHP Version: 7.2
### Description:
So in my `Commands/Kernel.php` i have the following code which is suppoused to emulate a supervisor for `queue:work`:
```
$schedule->command('queue:restart')->everyFiveMinutes();
$schedule->command('queue:work')->name('queue_work_name')->withoutOverlapping()->everyMinute()->runInBackground();
```
I've already cleared every tipe of cache, however in my process list using `ps aux | grep "php artisan"` I see this:
```
marcober 20527  0.0  0.0 376636 30956 ?        S    15:52   0:00 /opt/alt/php72/usr/bin/php artisan queue:work
marcober 22397  0.1  0.0 376636 30952 ?        S    15:53   0:00 /opt/alt/php72/usr/bin/php artisan queue:work
marcober 24157  0.1  0.0 376636 30956 ?        S    15:54   0:00 /opt/alt/php72/usr/bin/php artisan queue:work
marcober 29198  0.7  0.0 376636 30956 ?        S    15:56   0:00 /opt/alt/php72/usr/bin/php artisan queue:work
```
Why are they running together if i have `->withoutOverlapping()`?
PS: my only cron job is the one that you can find on the documentation 

### Steps To Reproduce:
add this to the Kernel:
```
$schedule->command('queue:restart')->everyFiveMinutes();
$schedule->command('queue:work')->name('queue_work_name')->withoutOverlapping()->everyMinute()->runInBackground();
```
and this cron job:
```
*  *  *  *  *  path/to/my/project && php artisan schedule:run >> /dev/null 2>&1
```

**Comment by vladislavtkachenko** [comment]
You need to delete "everyMinute" method.

**Comment by AlbertoSinigaglia** [comment]
> You need to delete "everyMinute" method.

@vladyslavstartsev  I've already tried, but nothing has changed, I still see a lot of instances of `queue:work` running together

**Comment by driesvints** [comment]
Moving to discussions.

**Comment by AlbertoSinigaglia** [comment]
I was actually looking at Laravel Frawork code and i can see that it use a file name for the lock like `schedule-*sha1 of the name*` inside `storage/framework`, but `ls -la | grep "schedule"` in that folder shows nothing

**Comment by AlbertoSinigaglia** [comment, marked as answer]
https://github.com/laravel/framework/issues/18202#issuecomment-387621069 I've just read this, and i'd remembered that I had `Artisan::call('cache:clear')` in my `schedule` method as first line... everything makes sense now, it was that line that was removing the lock every time...

**Reply by JenuelDev** [reply]
wait what did you do? You mean your clearning the cache so that is why your scheduler is not working?

**Reply by ghost** [reply]
He cleared the lock that withoutOverlapping was using (from the cache)

**Reply by ghost** [reply]
@AlbertoSinigaglia bear in mind that if you use log to stdout you loose any log from background commands, in this case also fro the worker.
